include/ddc/is_uniform_domain_v

```diff
@@ -0,0 +1,3 @@
+// SPDX-License-Identifier: MIT
```

Comment:
There is the header `UniformDomain`, if we change that, we also need to change `RectilinearDomain`

Reply:
I think we should and we should clarify that the file names should match symbol names

Reply:
But what symbols ? some of our public symbols don't have headers, i am afraid it will be difficult to respect in practice.

Reply:
I believe we should do it for all "public" symbols, those we expect our users to manipulate